lib/script.ts

```diff
@@ -32,28 +56,32 @@ export default class Script {
       options.readOnly = true;
     }
 
-    const evalsha = new Command("evalsha", [this.sha].concat(args), options);
+    const evalsha = new this.Command(
+      "evalsha",
+      [this.sha].concat(args),
+      options
+    );
     evalsha.isCustomCommand = true;
 
-    const result = container.sendCommand(evalsha);
-    if (isPromise(result)) {
-      return asCallback(
-        result.catch((err: Error) => {
-          if (err.toString().indexOf("NOSCRIPT") === -1) {
-            throw err;
-          }
-          return container.sendCommand(
-            new Command("eval", [this.lua].concat(args), options)
-          );
-        }),
-        callback
+    evalsha.promise = evalsha.promise.catch((err: Error) => {
+      if (err.toString().indexOf("NOSCRIPT") === -1) {
+        throw err;
+      }
+
+      // Resend the same custom evalsha command that gets transformed to an eval
+      // in case it's not loaded yet on the connectionDo an eval as fallback, redis will hash and load it
+      const resend = new this.Command(
```

Comment:
We could remove this fallback if order of delivery needs to be guaranteed.
It will still clear the cache for the next command that gets sent to redis.
https://github.com/luin/ioredis/pull/1499/files#diff-3618aaced6e025d978d2cdde354e756687916385e9024868d0ad3fff887c55c6R24

Reply:
I'm thinking to include this breaking change in the next major version https://github.com/luin/ioredis/pull/1504 so we can just retry custom commands in pipeline as well.